### PR TITLE
feat(brew): add alias subcommand

### LINKED ISSUE #1790: [brew] - missing subcommand `alias`
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

brew

### Which statement makes the most sense?

This completion spec isn't complete

### Issue Details

`alias` is a subcommand that is not an inbuilt subcommand of `brew` but can be added using `brew tap homebrew/aliases,` which provides the ability to create homebrew aliases similar to git aliases.

Ref: [Homebrew Aliases](https://github.com/Homebrew/homebrew-aliases)

```shell
$ brew alias --help                                                                                                                            
Usage: brew alias [alias ... | alias=command]

Show existing aliases. If no aliases are given, print the whole list.

      --edit                       Edit aliases in a text editor. Either one or
                                   all aliases may be opened at once. If the
                                   given alias doesn't exist it'll be
                                   pre-populated with a template.
  -d, --debug                      Display any debugging information.
  -q, --quiet                      Make some output more quiet.
  -v, --verbose                    Make some output more verbose.
  -h, --help                       Show this message.
```

### PR DESCRIPTION
**Pull request by seenukarthi**
This PR adds `alias` subcommand to brew.

Closes #1790

### CODE REVIEW

File: src/brew.ts [hunk 1]
```diff
@@ -67,6 +67,19 @@ const generateAllCasks: Fig.Generator = {
...
+    return out
+      .split("\n")
+      .filter((line) => line && line.trim() !== "")
+      .map((line) => ({
+        name: line,
+        icon: "fig://icon?type=command",
+        description: `Execute alias ${line}`,
+      }));
+  },
+});
```
Comment (fedeci) [inline src/brew.ts:82]:
```suggestion
const generatorAlias: Fig.Generator = {
  script: 'find ~/.brew-aliases/ -type f ! -name "*.*" -d 1 | sed "s/.*\\///"',
  postProcess: function (out) {
    return out
      .split("\n")
      .filter((line) => line && line.trim() !== "")
      .map((line) => ({
        name: line,
        icon: "fig://icon?type=command",
        description: `Execute alias ${line}`,
      }));
  },
};
```

File: src/brew.ts [hunk 2]
```diff
@@ -1576,13 +1589,51 @@ const completionSpec: Fig.Spec = {
...
+          description: "Make some output more verbose",
+        },
+        {
+          name: ["-h", "--help"],
+          description: "Show help message",
+        },
+      ],
+      args: {
+        name: "alias",
+        generators: generatorAlias(),
```
Comment (fedeci) [inline src/brew.ts:1619]:
```suggestion
        generators: generatorAlias,
```

File: src/brew.ts [hunk 3]
```diff
@@ -1576,13 +1589,51 @@ const completionSpec: Fig.Spec = {
...
   ],
   options: [
     {
       name: "--version",
       description: "The current Homebrew version",
     },
   ],
+  args: {
+    name: "alias",
+    generators: generatorAlias(),
```
Comment (fedeci) [inline src/brew.ts:1633]:
```suggestion
    generators: generatorAlias,
```

**Review by fedeci — CHANGES_REQUESTED** [review]

File: src/brew.ts [hunk 4]
```diff
@@ -67,6 +67,19 @@ const generateAllCasks: Fig.Generator = {
     }));
   },
 };
+const generatorAlias: Fig.Generator = {
```
Comment (AlanSyue) [inline src/brew.ts:70]: [nit, discussion]
Wouldn't this name be more concise?
```suggestion
const alias: Fig.Generator = {
```

**Review by AlanSyue — APPROVED** [review]

File: src/brew.ts (hunk 4 above)
Comment (seenukarthi) [reply src/brew.ts:70]: @AlanSyue I was following the convention in the file.

May be I could change is to `generateAlias` instead of `generatorAlias`

Comment (AlanSyue) [reply src/brew.ts:70]: I think `generateAliases` is nice 😀

**Review by AlanSyue — APPROVED** [review]

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**